packages/api-security/src/createSecurity/groupsTeamsPlugins/listGroupsFromPlugins.ts

```diff
@@ -0,0 +1,45 @@
+import { SecurityConfig } from "~/types";
+
+export interface ListGroupsFromPluginsParams {
+    listGroupsFromPluginsCallback?: SecurityConfig["listGroupsFromPluginsCallback"];
+    where: {
+        tenant: string;
+        id_in?: string[];
+        slug_in?: string[];
+    };
+}
+
+export const listGroupsFromPlugins = (params: ListGroupsFromPluginsParams) => {
+    const { listGroupsFromPluginsCallback, where } = params;
+    if (!listGroupsFromPluginsCallback) {
+        return [];
+    }
+
+    const groups = listGroupsFromPluginsCallback().map(plugin => {
+        return plugin.securityRole;
+    });
```

Comment:
This would be part of the `groupsProvider` implementation, since `plugin.securityRole` is an implementation detail.

Reply:
Done.